sql/mozfun/norm/get_earliest_value/udf.sql

```diff
@@ -0,0 +1,65 @@
+CREATE OR REPLACE FUNCTION norm.get_earliest_value(
+  value_set ARRAY<STRUCT<value STRING, value_date DATETIME>>
+)
+RETURNS STRUCT<earliest_value STRING, earliest_date DATETIME> AS (
+  (
+    WITH unnested AS (
+      SELECT
+        *
+      FROM
+        UNNEST(value_set)
+        WITH OFFSET AS offset
+      ORDER BY
+        offset ASC
+    )
+    SELECT
+      ARRAY_AGG(STRUCT(value, value_date) ORDER BY value_date ASC)[SAFE_OFFSET(0)]
+    FROM
+      unnested
+    WHERE
+      value IS NOT NULL
+      AND value_date IS NOT NULL
```

Comment:
This value is not expected to be null as pings have a timestamp, so it's meant to avoid returning values without dates.